homeassistant/components/blue_current/__init__.py

```diff
@@ -67,6 +79,47 @@ async def async_setup_entry(
     return True
 
 
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up Blue Current."""
+
+    async def start_charge_session(service_call: ServiceCall) -> None:
+        """Start a charge session with the provided device and charge card ID."""
+        # When no charge card is provided, use the default charge card set in the config flow.
+        charging_card_id = service_call.data.get(CHARGING_CARD_ID)
+        device_id = service_call.data.get(CONF_DEVICE_ID)
+
+        if device_id is None:
+            raise ServiceValidationError
+
+        # Get the device based on the given device ID.
+        device = dr.async_get(hass).devices.get(device_id)
+
+        if device is None:
+            raise ServiceValidationError
+
+        config_entry = hass.config_entries.async_get_entry(
+            list(device.config_entries)[0]
+        )
+
+        if config_entry is None or config_entry.state is not ConfigEntryState.LOADED:
+            raise ServiceValidationError
+
+        connector = config_entry.runtime_data
+
+        # Get the evse_id from the identifier of the device.
+        evse_id = list(device.identifiers)[0][1]
+
+        # When no charging card is provided, use no charging card (BCU_APP = no charging card).
+        if charging_card_id is None:
+            charging_card_id = BCU_APP
+
+        await connector.client.start_session(evse_id, charging_card_id)
+
+    hass.services.async_register(DOMAIN, START_CHARGE_SESSION, start_charge_session)
```

Comment:
Make sure you supply a schema. This way you don't have to check if device_id None for example